addons/mail/static/src/components/composer/tests/composer_tests.js

```diff
@@ -278,21 +270,19 @@ QUnit.test('add emoji replaces (keyboard) text selection', async function (asser
 
     // simulate selection of all the content by keyboard
     composerTextInputTextArea.setSelectionRange(0, composerTextInputTextArea.value.length);
-
-    // select emoji
+    await afterNextRender(
+        () => document.querySelector(`.o_ComposerTextInput_textarea`).dispatchEvent(new window.KeyboardEvent('focusout'))
+    );
```

Comment:
strange to need to add this

Reply:
Yeah, I supposed click on emojis button would focus out the textarea but apparently it doesn't.
Here the `focusout` is to ensure textarea content & selection is saved in models.
Without this, the previous selection (select all) is not saved, so selection is still at end of content, thus resulting in bad assert (emoji put at end of string, instead of replace).

Also the test simulates changing selection of textarea with `setSelectionRange`. While it does change the selection range in the textarea, it doesn't trigger any DOM event, so this doesn't save in models neither. In practice though, user always trigger a DOM event to change selection (`click`/`keydown`/`input`/etc.).

Reply:
Maybe legacy web.popover explicitly set focus on popover content on mount, which did trigger a `focusout` even on simulated click?